modules/ROOT/pages/java.adoc

```diff
@@ -842,6 +842,154 @@ The level of compression is set on Sync Gateway and can be tuned in the configur
 To optimize for speed, the replication protocol doesn't guarantee that documents will be received in a particular order.
 So we don't recommend to rely on that when using the replication or database change listeners for example.
 
+==== Replicator Notifications on a Custom Executor
+
+Prior to version 2.6, Couchbase Lite spun up multiple executors.
+This policy could result in too many threads being spun up.
+
+An executor manages a pool of threads and, perhaps, a queue in front of the executor to handle the asynchronous callbacks.
+Couchbase Lite API calls which are processed by an executor are listed below.
+
+[source]
+----
+Query.addChangeListener
+MessageEndpointListerner.addChangeListener
+LiveQuery.addChangeListener
+AbstractReplicator.addDocumentReplicationListener
+AbstractReplicator.addChangeListener
+Database.addChangeListener
+Database.addDocumentChangeListener
+Database.addDatabaseChangeListener
+Database.addChangeListener
+----
+
+As of version 2.6, Couchbase Lite spins up a single executor and a number of threads proportional to the number of CPUs `2 x CPUs`.
+
+Note that in some applications with other processes running in parallel, you may see `RejectedExecutionExceptions` error and the Couchbase Lite listener being aborted.
+This happens when there are more than `2 x CPUs` events being processed at the same time.
```

Comment:
This is factually incorrect.
Also, our exact policy is *NOT* something to which we want to commit in writing.
You probably should say something like:

Couchbase sometimes uses its own internal executor to run asynchronous client code.  While this is fine for small tasks, larger tasks -- those that take significant compute time, or that perform I/O -- can block Couchbase processing.  If this happens your application will fail with a `RejectedExecutionException` and it may be necessary to create a separate executor on which to run the large tasks.

Reply:
https://github.com/couchbase/docs-couchbase-lite/pull/136/commits/a5fcdae838dbdb792665e65712761e3d86c40af3 @bmeike Thanks for the re-worked text.